test/cypress/integration/stripe-setup/vat-number-company-number/vat_number_spec.js

```diff
@@ -0,0 +1,199 @@
+'use strict'
+
+// Local dependencies
+const commonStubs = require('../../../utils/common_stubs')
+const {
+  stubGetGatewayAccountStripeSetupSuccess,
+  stubGetStripeAccountSuccess,
+  stubGetGatewayAccountStripeSetupVatNumberCompanyNumberFlagChanged
+} = require('./support')
+
+describe('Stripe setup: VAT number page', () => {
+  const gatewayAccountId = 42
+  const userExternalId = 'userExternalId'
+
+  describe('Card gateway account', () => {
+    describe('when user is admin, account is Stripe and VAT number company number is not finished', () => {
+      beforeEach(() => {
+        cy.task('setupStubs', [
+          commonStubs.getUserStub(userExternalId, [gatewayAccountId]),
+          commonStubs.getGatewayAccountStub(gatewayAccountId, 'live', 'stripe'),
+          stubGetGatewayAccountStripeSetupSuccess(gatewayAccountId, false),
+          stubGetStripeAccountSuccess(gatewayAccountId, 'acct_123example123')
+        ])
+
+        cy.setEncryptedCookies(userExternalId, gatewayAccountId, {})
+
+        cy.visit('/vat-number-company-number/vat-number')
+      })
+
+      it('should display page correctly', () => {
+        cy.get('h1').should('contain', 'Add your organisation’s VAT number')
+
+        cy.get('#vat-number-form').should('exist')
+          .within(() => {
+            cy.get('input#vat-number-input').should('exist')
```

Comment:
I find asserting on the `name` of the form field can be quite nice because the unit tests for the submission to Stripe will probably start from some form data with named fields, which have to match those on the page. Having the names of the fields in a Cypress test sort of makes the names part of the contract (in that, if someone changes the names, the test will fail, which should at least make someone pause and think about whether they should be doing that).

Reply:
Good point 👍